sentry_streams/sentry_streams/adapters/arroyo/adapter.py

```diff
@@ -232,7 +233,20 @@ def broadcast(
         """
         Build a broadcast operator for the platform the adapter supports.
         """
-        raise NotImplementedError
+        assert (
+            stream.source in self.__consumers
+        ), f"Stream starting at source {stream.source} not found when adding a broadcast step"
+        self.__consumers[stream.source].add_step(BroadcastStep(route=stream, pipeline_step=step))
+
+        return {
+            route.name: Route(
+                source=stream.source,
```

Comment:
Just a general question but is my understanding correct that we track the Source as part of the Route so that we can commit offsets properly?

Reply:
The source in the route was not really for the consumer at runtime but for the adapter to be able to plug steps to the right stream.